thewalrus/quantum/fock_tensors.py

```diff
@@ -632,3 +635,62 @@ def tvd_cutoff_bounds(mu, cov, cutoff, hbar=2, check_is_valid_cov=True, rtol=1e-
         ps = np.real_if_close(np.diag(density_matrix(mu_red, cov_red, cutoff=cutoff, hbar=hbar)))
         bounds += 1 - np.cumsum(ps)
     return bounds
+
+def n_body_marginals(mean, cov, cutoff, n, hbar=2):
+    r"""Calculates the first n-body marginals of a Gaussian state.
+
+
+    For an M-mode Gaussian state there exists a photon number distribution with probability mass function
+    :math:`p[i_0,i_1,\ldots, i_{M-1}]`. The function ``n_body_marginals`` calculates the first n-body marginals
+    of the (all-mode) probability distribution :math:`p`. The :math:`n=1` marginals or single body marginals
+    are simply the probability that mode :math:`k` has :math:`i` photons, i.e. :math:`p_k[i]`.
+    For :math:`n=2` one obtains the two-body probabilities. For two modes :math:`k` and :math:`l` this is a
+    two dimensional probability distribution :math:`p_{k,l}[i,j]`.
+    Note that these marginals have interesting permutation properties, for example :math:`p_{k,l}[i,j] = p_{l,k}[j,i]`.
+
+    The function provided here takes advantage of these symmetries to minimize the amount of calculations.
+    The return of this function is a list of tensors where the first entry contains the one-body marginals of the :math:`M` modes
+    (giving a tensor of shape ``(M, cutoff)``), the second entry contains the two-body marginals (giving a tensor of shape ``(M,M,cutoff, cutoff)``)
+    and so on and so forth.
+
+    To be clever about not calculating things that can be obtained by permutations it checks whether the index vector representing the modes is sorted.
+    From the way `itertools.product` works we know that it will always produce a sorted index vector before generating any of its unordered permutations.
```

Comment:
```suggestion
    From the way ``itertools.product`` works we know that it will always produce a sorted index vector before generating any of its unordered permutations.
```